File: backend/PetCare/Entities/VaccinationSchedule.cs
using PetCare.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace PetCare.Entities;

public class VaccinationSchedule
{
    [Key]
    public int Id { get; set; }

    [ForeignKey(nameof(Pet))]
    public int PetId { get; set; }

    [JsonIgnore]
    public Pet Pet { get; set; } = null!;

    [MaxLength(100)]
    public string VaccineName { get; set; } = "";

    public DateTime ScheduledDate { get; set; }
    public bool IsCompleted { get; set; } = false;

    public DateTime? CompletedAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/petcare/lib/models/vaccination_schedule.dart
class VaccinationSchedule {
  final int id;
  final int petId;
  final String vaccineName;
  final DateTime scheduledDate;
  final bool isCompleted;
  final DateTime? completedAt;
  final DateTime createdAt;

  const VaccinationSchedule({
    required this.id,
    required this.petId,
    required this.vaccineName,
    required this.scheduledDate,
    required this.isCompleted,
    this.completedAt,
    required this.createdAt,
  });

  factory VaccinationSchedule.fromJson(Map<String, dynamic> json) {
    return VaccinationSchedule(
      id: json['id'] as int,
      petId: json['petId'] as int,
      vaccineName: json['vaccineName'] as String,
      scheduledDate: DateTime.parse(json['scheduledDate']),
      isCompleted: json['isCompleted'] as bool,
      completedAt: json['completedAt'] != null ? DateTime.parse(json['completedAt']) : null,
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'petId': petId,
      'vaccineName': vaccineName,
      'scheduledDate': scheduledDate.toIso8601String(),
      'isCompleted': isCompleted,
      'completedAt': completedAt?.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
